### ISSUE TITLE
Empty comment In Select Tag Causes Problem With NVDA Screenreader

### ISSUE THREAD

**Issue by sharadJay** [emberjs/ember.js#12459]
I see that a empty comment is created inside select tag as first child(before options) like this
![image](https://cloud.githubusercontent.com/assets/5979667/10362093/ee053040-6dc9-11e5-8fcb-5af04724ebd7.png)
I investigated a bit on it and found it being created when both first and parent child is null in setNode function.

NVDA screenreader does not read out selected option as it gets confused with this comment. Can this comment be removed in some way?

**Comment by stefanpenner** [comment]
Whoa sounds like a bug in NVDA, although with some work we can likely work around it..

**Comment by sharadJay** [comment]
Wanted to check on significance of this comment actually, this can be removed if it doesn't serve any purpose

**Comment by krisselden** [comment]
@sharadJay we started using comments to hold empty dynamic slots where we used to use empty text nodes because of a chrome bug.

**Comment by krisselden** [comment]
Though it maybe the screen reader is just assuming the firstChild is an option not the firstElementChild

**Comment by sharadJay** [comment]
@krisselden yes this is the issue! Is there any way we can handle it ?

**Comment by rwjblue** [comment]
Can you please share a reproduction so we can see/play with this? 

Without an example, I would guess that there is some sort of conditional before the listing of options (this is pretty  common when using prompts) and if it is empty we emit a comment node. 

I thought that without an empty section, we shouldn't render an empty comment...

**Comment by krisselden** [comment]
@rwjblue it is the select components template that causes this, he does not need to reproduce I can confirm.

@sharadJay don't use the select component, use this: http://ember-twiddle.com/aae2320cf045c9ddc635

**Comment by rwjblue** [comment]
@krisselden - The last two stable releases of Ember do not include a default select implementation. I would like to confirm which version of Ember and what select implementation is actually being used.

**Comment by krisselden** [comment]
@rwjblue sounds good, the reason I said I can confirm because in his screenshot I can see the "ember-select" class name and I know the old select component has an #if block at the start of the options that is likely causing the issue.

**Comment by rwjblue** [comment]
Gotcha. Yeah, I assumed it was related to an if at the top for prompt (which is pretty common). Seems like that is the issue, and that it should be easy to work around by not using Ember.Select (which should die a horrible death).

**Comment by sharadJay** [comment]
Yes I see this is a good workaround. Thank you @krisselden
